src/BackwardCompatibility/SalesChannelServiceCompilerPass.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Omikron\FactFinder\Shopware6\BackwardCompatibility;
+
+use Omikron\FactFinder\Shopware6\Export\SalesChannelService;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+class SalesChannelServiceCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if ($container->has('Shopware\Core\System\SalesChannel\Context\CachedSalesChannelContextFactory')) {
+            $salesChannelService  = new Definition('Omikron\FactFinder\Shopware6\BackwardCompatibility\Extension\CachedSalesChannelContextFactory', [
```

Comment:
let's rename this variable to be more verbous about what it is about. It's not SalesChannelService but SalesChannelContextFactory

Reply:
sounds legit, will change it